app/models/user.rb

```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  before_save { self.email = email.downcase }  
+  validates :name, presence: true, length: { maximum: 50 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-]+(\.[a-z\d\-]+)*\.[a-z]+\z/i
```

Comment:
I like it that you also added a format validation, nice!

Rails also a regex for that, so you could use: `format: { with: URI::MailTo::EMAIL_REGEXP }`, but your implementation also works

Reply:
I will keep the current regex expression, because the one that is in the Rails library takes as valid the emails that do not have a TLD (e.g. adress@anythingwithoutadothere)